### PR TITLE
Add SelectLanguage Page

### LINKED ISSUE #251: Select Language Page
**Goals**

Screen Shot 2022-10-27 at 8 13 53 PM

We want users to be able to select the language that they want the app to be in when they open up the app for the first time. We can store the User's Language in [AsyncStorage](https://reactnative.dev/docs/asyncstorage) and refer to it whenever the user opens up the app a later time. Here is a breakdown of this issue:

1. Create the **Select Language Page** displayed in the screenshot above. Make sure to only display it when the user logins for the first time, and before the Landing Page. Also, make sure to persist the Language that the user chooses (`AsyncStorage` is your friend).
2. Whenever the app is opened at a later time, refer to that persisted language to set `i18n.locale` ([Expo Localization](https://docs.expo.dev/versions/latest/sdk/localization/)).

**Other Considerations**
Do we really need this page? Expo Localization has the following support:

```
// Set the locale once at the beginning of your app.
i18n.locale = Localization.locale;
```

Could we also make use of this when setting the default language on the **Select Language Page**.

### PR DESCRIPTION
**Pull request by eeshabarua**
## Description

Fixes #251 

## Screenshots
image

### CODE REVIEW

File: client/src/pages/SelectLanguage/SelectLanguage.js [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+WebBrowser.maybeCompleteAuthSession()
+
+const SelectLanguage = ({ navigation }) => {
+  const window = useWindowDimensions()
+
+  const [isEnglish, setIsEnglish] = useState(true)
+
+  const handlePressEnglish = () => {
+    // Alert.alert('Translating to English...')
```
Comment (ashayp22) [inline client/src/pages/SelectLanguage/SelectLanguage.js:55]: Remove Alert

File: client/src/pages/SelectLanguage/SelectLanguage.js [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    i18n.locale = 'en'
+    setIsEnglish(true)
+  }
+
+  const handlePressFrench = () => {
+    // Alert.alert('Translating to French...')
```
Comment (ashayp22) [inline client/src/pages/SelectLanguage/SelectLanguage.js:61]: Remove Alert

File: client/src/pages/SelectLanguage/SelectLanguage.js [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+  },
+})
+
+WebBrowser.maybeCompleteAuthSession()
+
+const SelectLanguage = ({ navigation }) => {
+  const window = useWindowDimensions()
+
+  const [isEnglish, setIsEnglish] = useState(true)
+
```
Comment (ashayp22) [inline client/src/pages/SelectLanguage/SelectLanguage.js:53]: Is this choice saved for the next time that the user opens the app?

File: client/src/utils/i18n/index.js [hunk 4]
```diff
@@ -1,16 +1,17 @@
...
 import { I18n } from 'i18n-js'
+import { NativeModules, Platform } from 'react-native'
 import translations from './translations'
 
 const i18n = new I18n(translations)
 
-// Set the locale once at the beginning of your app.
-// i18n.locale = Localization.locale
+/* This is a function to map the locale to an existing language on our languageData json object. i.e. fr_DZ -> fr */
+const convertLanguage = (locale) => (locale.indexOf('fr') >= 0 ? 'fr' : 'en')
```
Comment (ashayp22) [inline client/src/utils/i18n/index.js:8]: Clean function

File: client/src/navigator/Stacks/Stacks.js [hunk 5]
```diff
@@ -62,12 +63,19 @@ More reading: https://reactnavigation.org/docs/stack-navigator/
 
 export const AuthNavigator = () => (
   <AuthStack.Navigator
-    initialRouteName="Landing"
+    initialRouteName="SelectLanguage"
     headerMode="screen"
     screenOptions={{
       headerShown: false,
     }}
   >
```
Comment (ashayp22) [inline client/src/navigator/Stacks/Stacks.js:71]: Do we ensure that this screen is only shown when the user opens the app for the first time?

**Review by ashayp22 — CHANGES_REQUESTED** [review]
Great UI work Eesha, I left a few inline comments.

File: client/src/navigator/Stacks/Stacks.js (hunk 5 above)
Comment (eeshabarua) [reply client/src/navigator/Stacks/Stacks.js:80]: Yes, once logged in this page doesn't show up.

**Comment by eeshabarua** [comment]
NOTE: No current way to test Async Storage because dev lost the Log Out button

**Pull request merged**